### DISCUSSION TITLE
Organizing GitHub Repositories

### DISCUSSION THREAD

**Discussion by ghost** [pulsar-edit/.github#39] [General]
I feel like Pulsar`s GitHub repositories needs to be organized in one way or another.

Currently, to see what is relevant to a "category", you have to look inside the repository's README to find out if it's what you're looking for. And that is a bit tedious, as you have to read a part of a long or short document. Short documents are short, but they are still accounted for here.

There are some solutions that can be used to make things easier. All take a role in the documentation.

Here are some solutions:

- **Repository names having "namespaces".** This is an easy way of doing so, since it takes what the user sees first when searching for repositories relevant to their search term.
- **All READMEs with a "Category" badge or text.** I don't know if this really "works" for the user, since some just search them up and copy the links to paste into a description of a YouTube video or something. But we may as well try.

If any are applied, it may look better for organizing our things.

**Comment by confused-Techie** [comment]
We do currently have a [Org `REPOS.md`](https://github.com/pulsar-edit/.github/blob/main/REPOS.md) file, that attempts to accomplish this purely on a documentation side. While it is missing some things, as not everything had been moved over yet, I personally think it'd be cool to have something like this, but on the website, with a proper UI and such. 

Wouldn't help if you're looking through readmes like youre talking about, but could at least be a place to understand how everything fits together.

An alternative option, like you said a badge or text, I'm not sure how we could properly show those on GitHub, except maybe an emoji table. Where each emoji represents something like languages, ides, and core, that sorta thing.

**Reply by ghost** [reply]
There are badge websites like shields.io that can generate badges. Below every page on shields.io, you can generate custom badges. If you need them customized [here is some highlights on the documentation](https://shields.io/#:~:text=Styles,maxAge%22%20is%20also%20supported.) (link highlights available on Chrome only).

**Reply by Spiker985** [reply]
We currently use shields.io in the `pulsar` repo, and would likely use it everywhere else too

**Reply by confused-Techie** [reply]
> There are badge websites like shields.io that can generate badges. Below every page on shields.io, you can generate custom badges. If you need them customized [here is some highlights on the documentation](https://shields.io/#:~:text=Styles,maxAge%22%20is%20also%20supported.) (link highlights available on Chrome only).

Oh my bad, thought you meant badging when viewing multiple repos all at once. Like where you only see the repo name and description, that's where the idea of an emoji table came from.

But if we are counting on the user selecting each repo, then yeah the badging could absolutely be helpful, with a custom badge made for each general subject

**Reply by kaosine** [reply]
Yeah, badges was one of the things I meant to do today. It just didn't happen today, since I got distracted by several tasks I had on my mind and ended up getting maybe 1/3 of what I meant to do done...

**Comment by kaosine** [comment]
This is why we have all the teams we do. If you look at those, they're there. Granted, some need to be transferred from packages to documentation(as they are example code for tutorial content). Or they need to be archived because they're things we're dropping(see Etch related content that's going to be transferred to solidjs).

**Reply by ghost** [reply]
Teams are just visible to organization members. So they cannot access the teams. I'm meaning about how they can be accessible for anyone, not just organization members.
